### DISCUSSION TITLE
Open Wingdings.ttf font by STB_TrueType

### DISCUSSION THREAD

**Discussion by Jony01** [nothings/stb#1282] [Q&A]
I've trouble with initialisation of some fonts by function `stbtt_InitFont` which return zero

List of unopenable font on my windows:

"Bookshelf Symbol 7"
"Marlett"
"MS Outlook"
"MS Reference Specialty"
"MT Extra"
"Symbol"
"Webdings"
"Wingdings 2"
"Wingdings 3"
"Wingdings"
[fonts.zip](https://github.com/nothings/stb/files/7973062/fonts.zip)

Rest 270 true type fonts works fine.

Is there some restriction why I can't open this fonts?

![image](https://user-images.githubusercontent.com/15888762/151851493-956b736d-98f3-43e6-9dce-853e9aec2f36.png)

**Comment by nothings** [comment]
Most fonts contain information that tell software how to take a unicode (or ASCII) character and find the correct way to draw it. Because the primary API of stb_truetype is Unicode, we don't handle fonts that don't have that information.

All of the fonts in fonts.zip are "symbol" fonts which use a microsoft-specific mapping system from character identifiers (which aren't Unicode) to determine how to draw it. Because it's not unicode stb_truetype doesn't support it. I'm not sure why this has never come up before, given how long stb_truetype has been out.

Anyway, to fix it, we need to add support for cmap platformID=3 (Miicrosoft) encodingID = 0 (symbol font). Should probably make the logic first check for a non-symbol encoding, only check for 3/0 if it doesn't find a regular one first.

**Comment by nothings** [comment]
I've opened an issue, #1283
